cosmic-applet-status-area/src/components/app.rs

```diff
@@ -44,7 +45,143 @@ pub(crate) struct App {
     token_tx: Option<calloop::channel::Sender<TokenRequest>>,
 }
 
+static ICON_NAME_CACHE: OnceLock<Mutex<HashMap<String, Option<String>>>> = OnceLock::new();
+
 impl App {
+    fn find_flatpak_icon(search_term: &str) -> Option<String> {
+        let search_lower = search_term.to_lowercase();
+
+        let flatpak_dirs = [
+            format!(
+                "{}/.local/share/flatpak/exports/share/icons",
+                std::env::var("HOME").ok()?
+            ),
+            "/var/lib/flatpak/exports/share/icons".to_string(),
+        ];
```

Comment:
We should never need to hardcode these paths, since they are part of `XDG_DATA_DIRS`, which the `freedesktop-icons` crate already parses and checks correctly.

If we need to check additional variants like "_tray" for these icons, we should simply try to look these up with `freedesktop-icons` as well.